test/searchModal/test.js

```diff
@@ -222,34 +239,93 @@ define(['jquery', 'ui/searchModal', 'core/store', './mocks/searchStore'], functi
         });
     });
 
+    QUnit.module('class filter');
+    QUnit.test('class filter workflow is correct', function (assert) {
+        const ready = assert.async();
+        assert.expect(3);
+        const instance = searchModalFactory({
+            criterias: { search: 'example' },
+            url: '/test/searchModal/mocks/with-occurrences/search.json',
+            renderTo: '#testable-container',
+            rootClassUri: 'http://www.tao.lu/Ontologies/TAOItem.rdf#Item',
+            testMocks: { classTree }
+        });
+
+        instance.on('ready', function () {
+            const $container = $('.search-modal');
+            const classInput = $container.find('.class-filter');
+            const clearButton = $('.btn-clear');
+            const QtiInteractionsClassNode = $('.class-tree [title="QTI Interactions"]');
```

Comment:
jQuery variables must be prefixed with a `$`
Variable names must not start with an upper case, this is reserved for class names.